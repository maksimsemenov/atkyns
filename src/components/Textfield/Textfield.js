import React, { PropTypes } from 'react'
import Autocomplete from 'react-autocomplete'
import './Textfield.less'


// Standart helper functions for autocomplete
export const renderMenu = (items, value, style) =>
  items ?
  <ul className='textfield-ac' children={items} style={style} /> :
  null

export const renderItem = (item, isHighlighted) =>
  <li
    key={item.value}
    className={`textfield-acItem${isHighlighted ? 'is-highlighted' : ''}`}
  >{item.name}</li>

export const getItemValue = item => item.name
export const shouldItemRender = (item, value) =>
  item.value.toLowerCase().search(value.toLowerCase()) !== -1

const defaultAC = {
  renderItem,
  renderMenu,
  getItemValue,
  shouldItemRender
}

// Textfield component itself
const Textfield = ({
  value,
  mask,
  wrapperStyle,
  appearance,
  autocomplete,
  onChange,
  error,
  note,
  ...rest
}) => {
  const maskedValue = mask ? mask(value) : value
  const ac = autocomplete ? {...defaultAC, ...autocomplete } : undefined
  const inputProps = {
    className: `textfield${error ? ' has-error' : ''}`,
    ...rest
  }
  return (
    <div className='textfield__control' style={wrapperStyle}>
      {ac ? (
        <Autocomplete
          value={maskedValue}
          onSelect={onChange}
          onChange={(event, value) => onChange(value)}
          inputProps={inputProps}
          {...ac}
        />
      ) : (
        <input value={maskedValue} onChange={onChange} {...inputProps} />
      )}
      {error && <div className='ob-textfield-error'>{error}</div>}
      {!error && note && <div className='ob-textfield-note'>{note}</div>}
    </div>
  )
}

Textfield.propTypes = {
  value: PropTypes.string,
  type: PropTypes.oneOf(['text', 'email', 'password', 'url', 'date', 'number']),
  wrapperStyle: PropTypes.object,
  autocomplete: PropTypes.shape({
    items: PropTypes.arrayOf(PropTypes.shape({
      value: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    })).isRequired,
    shouldItemRender: PropTypes.func
  }),
  mask: PropTypes.func,
  onChange: PropTypes.func.isRequired,

  appearance: PropTypes.oneOf(['default', 'inGroup']),
}
Textfield.defaultProps = {
  type: 'text',
}

export default Textfield
