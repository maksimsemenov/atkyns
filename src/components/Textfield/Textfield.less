@import '~constants/styles.less';

.textfield {
  width: 100%;
  box-sizing: border-box;
  padding: 10px 10px 12px;
  border-radius: 2px;
  border: 1px solid @color-grey;
  font-size: @font-size-form;
  font-weight: @font-weight-basic;

  transition: border-color @transition-basic;

  &:hover {
    border-color: @color-dark-grey;
    transition: border-color @transition-active;
  }
  &:active,
  &:focus {
    border-color: @color-accent;
    outline: none;
    box-shadow: 0 0 3px 0 fade(@color-accent, 30%);
    transition: border-color @transition-active;
  }

  &::placeholder {
    color: @color-grey;
    font-weight: @font-weight-basic;
  }
}
