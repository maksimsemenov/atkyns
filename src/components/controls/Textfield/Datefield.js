import React, { Component, PropTypes } from 'react'
import l from 'utils/local'

const months = [
  l('January'),
  l('Febrary'),
  l('March'),
  l('April'),
  l('May'),
  l('June'),
  l('July'),
  l('August'),
  l('September'),
  l('October'),
  l('November'),
  l('December'),
]
const weekDays = [
  l('Mo'),
  l('Tu'),
  l('We'),
  l('Th'),
  l('Fr'),
  l('Sa'),
  l('Su')
]

/*
 * Calendar
 */

class Calendar extends Component {
  constructor(props) {
    super(props)
    const today = new Date()
    this.state = {
      mode: 'month',
      day: today.getDate(),
      month: today.getMonth(),
      year: today.getFullYear(),
      decade: today.getFullYear() - 4
    }
  }

  handleMonthClick() {
    this.setState({ mode: 'year'})
  }
  handleYearClick() {
    this.setState({ mode: 'decade'})
  }
  handleMonthItemClick(month) {
    this.setState({ month, mode: 'month' })
  }
  handleYearItemClick(year) {
    this.setState({ year, mode: 'month' })
  }
  handleNextClick() {
    switch (this.state.mode) {
      case 'month':
        this.setState(prevState => {
        if (prevState.month === 11) {
          return { month: 0, year: prevState.year + 1 }
        }
          return { month: prevState.month + 1 }
        })
        break
      case 'year':
        this.setState(prevState => ({ year: prevState.year + 1}))
        break
      default:
        this.setState(prevState => ({ decade: prevState.decade + 9}))
    }
  }
  handlePrevClick() {
    switch (this.state.mode) {
      case 'month':
        this.setState(prevState => {
        if (prevState.month === 0) {
          return { month: 11, year: prevState.year - 1 }
        }
          return { month: prevState.month - 1 }
        })
        break
      case 'year':
        this.setState(prevState => ({ year: prevState.year - 1}))
        break
      default:
        this.setState(prevState => ({ decade: prevState.decade - 9}))
    }
  }

  renderYears() {
    return new Array(9).fill(1).map((item, index) =>
      <div
        key={index}
        className='textfield__calendarItem textfield__calendarItem--year'
        onClick={() => this.handleYearItemClick(this.state.decade + index)}
        onMouseDown={this.mouseDown}
      >{this.state.decade + index}</div>
    )
  }
  renderMonths() {
    return months.map((month, index) =>
      <div
        key={index}
        className='textfield__calendarItem textfield__calendarItem--month'
        onClick={() => this.handleMonthItemClick(index)}
        onMouseDown={this.mouseDown}
      >{month}</div>
    )
  }
  renderWeekDayNames() {
    return weekDays.map((day, index) =>
      <div key={index} className='textfield__calendarWeekDay'>{day}</div>
  )}

  renderWeek ({ children, key }) {
    return <div key={key} className='textfield__calendarWeek'>{children}</div>
  }
  renderDay (date, key, onClick, onMouseDown) {
    return <div
      key={key}
      className='textfield__calendarDay'
      onClick={() => onClick(date)}
      onMouseDown={() => onMouseDown()}
    />
  }

  renderDays(items) {
    const { year, month } = this.state
    let startDate = new Date(year, month, 1)
    let monthArray = [this.renderWeek({ children: this.renderWeekDayNames(), key: -1 })]
    for (let i = 0; i < 7; i++) {
      if (startDate.getMonth() === month) {
        let week = []
        for (let j = 0; j < 7; j++) {
          const day = startDate.getDay()
          if ((day - 1 === j || (day === 0 && j === 6)) && startDate.getMonth() === month) {
            const monthDay = startDate.getDate()
            week.push(this.renderDay(startDate, j, this.props.onSelect, this.props.onMouseDown))
            startDate = new Date(year, month, monthDay + 1)
          } else {
            week.push(<div key={j} className='textfield__calendarDay is-empty' />)
          }
        }
        monthArray.push(this.renderWeek({ children: week, key: i }))
      }
    }
    return monthArray
  }

 render() {
   const { month, year, mode } = this.state
   return (
     <div className='textfield__calendar' style={this.props.style}>
       <div className='textfield__calendarHeader'>
         <div
           className='textfield__calendarPrev'
           onClick={() => this.handlePrevClick()}
           onMouseDown={this.mouseDown}
         ><div className='textfield__calendarPrevSign' /></div>
         <div
           className='textfield__calendarMonthTitle'
           onClick={() => this.handleMonthClick()}
           onMouseDown={this.mouseDown}
         >
           {this.months[month]}
         </div>
         <div
           className='textfield__calendarYearTitle'
           onClick={() => this.handleYearClick()}
           onMouseDown={this.mouseDown}
         >
           {year}
         </div>
         <div
           className='textfield__calendarNext'
           onClick={() => this.handleNextClick()}
           onMouseDown={this.mouseDown}
         ><div className='textfield__calendarNextSign' /></div>
       </div>
       <div className='textfield__calendarBody'>
         {
           mode === 'decade' ? this.renderYears() :
           mode === 'year' ? this.renderMonths() :
           this.renderDays(this.props.items)
         }
       </div>
     </div>
   )
 }
}

Calendar.propTypes = {
  onSelect: PropTypes.func.isRequired,
  onMouseDown: PropTypes.func.isRequired
}



 /*
 * Datafield
 */

class Datefield extends Component {
  constructor(props) {
    super(props)
    this.state({
      isOpen: false,
    })
    this.selectItemFromMouse = this.selectItemFromMouse.bind(this)
  }

  componentWillMount () {
    this._ignoreBlur = false
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.value !== '') {
      this.setState({ isOpen: true })
    } else {
      this.setState({ isOpen: false })
    }
  }

  componentDidUpdate (prevProps, prevState) {
    if (this.state.isOpen === true && prevState.isOpen === false)
      this.setMenuPositions()
  }

  handleKeyDown (event) {
    if (this.keyDownHandlers[event.key])
      this.keyDownHandlers[event.key].call(this, event)
    else {
      this.setState({
        isOpen: true
      })
    }
  }

  handleChange (event) {
    this.props.onChange(event, event.target.value)
  }

  keyDownHandlers = {
    Escape (event) {
      this.setState({
        isOpen: false
      })
    }
  }

  setMenuPositions () {
    var node = this.refs.input
    var rect = node.getBoundingClientRect()
    var computedStyle = global.window.getComputedStyle(node)
    var marginBottom = parseInt(computedStyle.marginBottom, 10) || 0;
    var marginLeft = parseInt(computedStyle.marginLeft, 10) || 0;
    var marginRight = parseInt(computedStyle.marginRight, 10) || 0;
    this.setState({
      menuTop: rect.bottom + marginBottom,
      menuLeft: rect.left + marginLeft,
      menuWidth: rect.width + marginLeft + marginRight
    })
  }

  selectItemFromMouse (item) {
    var value = item.value
    this.setState({
      isOpen: false,
    }, () => {
      this.props.onSelect(value, item)
      this.refs.input.focus()
    })
  }

  setIgnoreBlur (ignore) {
    this._ignoreBlur = ignore
  }

  renderMenu () {
    return (
      <Calendar
        onMouseDown={() => this.setIgnoreBlur(true)}
        onSelect={this.selectItemFromMouse}
      />
    )
  }

  handleInputBlur () {
    if (this._ignoreBlur) return
    this.setState({ isOpen: false })
  }

  handleInputFocus () {
    if (this._ignoreBlur) {
      this.setIgnoreBlur(false)
      return
    }
    // We don't want `selectItemFromMouse` to trigger when
    // the user clicks into the input to focus it, so set this
    // flag to cancel out the logic in `handleInputClick`.
    // The event order is:  MouseDown -> Focus -> MouseUp -> Click
    this._ignoreClick = true
    this.setState({ isOpen: true })
  }

  isInputFocused () {
    var el = this.refs.input
    return el.ownerDocument && (el === el.ownerDocument.activeElement)
  }

  handleInputClick () {
    // Input will not be focused if it's disabled
    if (this.isInputFocused() && this.state.isOpen === false)
      this.setState({ isOpen: true })
    else if (this.state.highlightedIndex !== null && !this._ignoreClick)
      this.selectItemFromMouse(this.getFilteredItems()[this.state.highlightedIndex])
    this._ignoreClick = false
  }

  render () {
    return (
      <div style={{...this.props.wrapperStyle}}>
        <input
          {...this.props.inputProps}
          role="combobox"
          aria-autocomplete="list"
          autoComplete="off"
          ref="input"
          onFocus={this.handleInputFocus}
          onBlur={this.handleInputBlur}
          onChange={(event) => this.handleChange(event)}
          onKeyDown={(event) => this.handleKeyDown(event)}
          onKeyUp={(event) => this.handleKeyUp(event)}
          onClick={this.handleInputClick}
          value={this.props.value}
        />
        {('open' in this.props ? this.props.open : this.state.isOpen) && this.renderMenu()}
      </div>
    )
  }
}

Datefield.propTypes = {
  value: PropTypes.any,
  onChange: PropTypes.func,
  inputProps: PropTypes.object,
  wrapperStyle: PropTypes.object,
  onMenuVisibilityChange: PropTypes.func,
  open: PropTypes.bool,
}

export default Datefield
