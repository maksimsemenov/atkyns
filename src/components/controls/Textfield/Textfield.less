@import '~constants/styles.less';

.textfield__control {
  width: 350px;
  max-width: 100%;

  .stage__fieldRow & {
    width: auto;
    flex-shrink: 0;
  }
}
.textfield {
  position: relative;
  z-index: 1;
  width: 100%;
  box-sizing: border-box;
  padding: 10px 10px 12px;
  border-radius: @radius-control;
  border: 1px solid @color-gray;
  font-size: @font-size-form;
  font-weight: @font-weight-basic;

  // transition: border-color @transition-basic, z-index @transition-basic;

  &:hover {
    border-color: @color-dark-gray;
    // transition: border-color @transition-active, z-index @transition-active-step;
  }
  &:active,
  &:focus {
    border-color: @color-accent;
    outline: none;
    box-shadow: 0 0 3px 0 fade(@color-accent, 30%);
    // transition: border-color @transition-active, z-index @transition-active-step;
  }

  &::placeholder {
    color: @color-gray;
    font-weight: @font-weight-basic;
  }

  &.has-error {
    border-color: @color-red;
    box-shadow: 0 0 3px 0 fade(@color-red, 30%);
  }
  .stage__fieldColumn &,
  .stage__fieldRow & {
    border-radius: 0;

    &:hover {
      z-index: 2;
    }
    &:active,
    &:focus {
      z-index: 3;
    }
  }
  .stage__fieldRow.is-first &.is-first {
    border-top-left-radius: @radius-control;
  }
  .stage__fieldRow.is-last &.is-first {
    border-bottom-left-radius: @radius-control;
  }
  .stage__fieldRow.is-first &.is-last {
    border-top-right-radius: @radius-control;
  }
  .stage__fieldRow.is-last &.is-last {
    border-bottom-right-radius: @radius-control;
  }
}

.textfield__error {
  padding: 2px;
  line-height: 1.4;
  color: @color-red;
  font-size: @font-size-small;
}
.textfield__note {
  padding: 2px;
  line-height: 1.4;
  color: @font-color-light;
  font-size: @font-size-small;
}


//
// .textfield__calendar {
//   width: 100%;
//   max-width: 300px;
//   padding: 20px 25px;
//   box-sizing: border-box;
//   display: flex;
//   flex-direction: column;
//
//   background-color: #fff;
//   border-radius: 2px;
//   box-shadow: 0 1px 3px 0 fade(#000, 20%);
// }
// .textfield__calendarHeader {
//   display: flex;
//   flex-direction: row;
//   flex-wrap: nowrap;
//   align-items: center;
//   margin-bottom: 10px;
// }
//
// .textfield__calendarMonthTitle {
//   padding: 5px;
//   margin: 0 5px;
//   cursor: pointer;
// }
// .textfield__calendarYearTitle {
//   padding: 5px;
//   margin: 0 5px;
//   cursor: pointer;
// }
//
// .textfield__calendarWeek {
//   display: flex;
//   flex-direction: row;
//   flex-wrap: nowrap;
// }
//
// .textfield__calendarWeekDay {
//   position: relative;
//   box-sizing: border-box;
//   width: 14.2857%;
//   padding: 5px;
//   margin-bottom: 5px;
//
//   text-align: center;
//   font-size: @font-size-xsmall;
//   font-weight: @font-weight-basic;
//   color: @font-color-light;
//   letter-spacing: 0.04em;
// }
//
// .textfield__calendarDay {
//   position: relative;
//   width: 14.2857%;
//   height: 34px;
//   text-align: center;
//   padding: 5px auto;
//   cursor: pointer;
//   margin: 2px 0;
//
//   &::before {
//     content: '';
//     position: absolute;
//     z-index: 0;
//     top: 50%;
//     left: 50%;
//     width: 34px;
//     height: 34px;
//     transform: translate(-50%, -50%) scale(0);
//     border-radius: 50%;
//     opacity: 0;
//     background-color: @color-light-blue;
//     transition: transform @transition-basic, opacity @transition-basic;
//   }
//
//   &:hover::before {
//     opacity: 1;
//     transform: translate(-50%, -50%) scale(1);
//     transition: transform @transition-active, opacity @transition-active;
//   }
//
//   &.is-empty {
//     pointer-events: none;
//
//     &::before {
//       content: none;
//     }
//   }
// }
// .textfield__calendarDayText {
//   position: relative;
//   z-index: 1;
// }


// .calendar__prev,
// .calendar__next {
//   position: relative;
//   width: 30px;
//   height: 30px;
//   background-color: transparent;
//   border: none;
//
//   &::before {
//     content: '';
//     position: absolute;
//     z-index: 0;
//     width: 100%;
//     height: 100%;
//     top: 50%;
//     left: 50%;
//     transform: translate(-50%, -50%) scale(0);
//     border-radius: 50%;
//     border: 1px solid @color-bg-light-gray;
//     opacity: 0;
//     background-color: @color-bg-light-gray;
//     transition: @transition-basic;
//   }
//   &:hover::before {
//     transform: translate(-50%, -50%) scale(1);
//     opacity: 1;
//   }
//   &:active::before {
//     background-color: @color-border-light-gray;
//     border-color: @color-border-light-gray;
//   }
//   &.is-disabled {
//     opacity: @opacity-disabled;
//     pointer-events: none;
//   }
// }
// .calendar__prev {
//   left: -5px;
// }
// .calendar__next {
//   right: -5px;
// }
// .calendar__prev-sign {
//   position: relative;
//   z-index: 1;
//   width: 100%;
//   height: 100%;
//
//   &::before,
//   &::after {
//     content: '';
//     position: absolute;
//     left: 50%;
//     top: 50%;
//     width: 1px;
//     height: 10px;
//     background-color: @color-font-light;
//     transition: @transition-basic;
//   }
//   &::before {
//     transform: translate(-5px, -100%) rotate(50deg);
//     transform-origin: left bottom;
//   }
//   &::after {
//     transform: translate(-5px, 0) rotate(-50deg);
//     transform-origin: left top;
//   }
//
//   .calendar__prev:hover &,
//   .calendar__prev:active & {
//     &::before,
//     &::after {
//       background-color: @color-font-medium;
//     }
//   }
// }
// .calendar__next-sign {
//   position: relative;
//   z-index: 1;
//   width: 100%;
//   height: 100%;
//
//   &::before,
//   &::after {
//     content: '';
//     position: absolute;
//     left: 50%;
//     top: 50%;
//     width: 1px;
//     height: 10px;
//     background-color: @color-font-light;
//     transition: @transition-basic;
//   }
//   &::before {
//     transform: translate(4px, -100%) rotate(-50deg);
//     transform-origin: right bottom;
//   }
//   &::after {
//     transform: translate(4px, 0) rotate(50deg);
//     transform-origin: right top;
//   }
//
//   .calendar__next:hover &,
//   .calendar__next:active & {
//     &::before,
//     &::after {
//       background-color: @color-font-medium;
//     }
//   }
// }
//
// .calendar__day-names {
//   position: relative;
//   box-sizing: border-box;
//   width: 100%;
//
//   display: flex;
//   flex-direction: row;
//   flex-wrap: nowrap;
//   align-items: center;
//   justify-content: center;
// }
// .calendar__day-name {
//   position: relative;
//   box-sizing: border-box;
//   width: 14.2857%;
//   padding: 5px;
//
//   text-align: center;
//   font-size: @font-size-xsmall;
//   font-weight: @font-weight-medium;
//   color: @color-font-light;
//   letter-spacing: 0.04em;
//
//   &:first-child {
//     margin-left: -7.1428%;
//   }
// }
//
// .calendar__week {
//   position: relative;
//   display: flex;
//   flex-direction: row;
//   flex-wrap: nowrap;
//   align-items: center;
//   justify-content: center;
// }
// .calendar__day {
//   position: relative;
//   z-index: 0;
//   box-sizing: border-box;
//   width: 14.2857%;
//   padding: 10px 10px 8px 10px;
//
//   text-align: center;
//   font-size: @font-size-basic;
//   font-weight: @font-weight-light;
//   color: @color-font-basic;
//   line-height: 1;
//   cursor: pointer;
//   background-color: transparent;
//
//   &::before {
//     content: '';
//     position: absolute;
//     z-index: -1;
//     top: 50%;
//     left: 50%;
//     transform: translate(-50%, -50%) scale(0);
//     width: 28px;
//     height: 28px;
//     border-radius: 50%;
//     border: 1px solid @color-bg-light-gray;
//     opacity: 0;
//     background-color: @color-bg-light-gray;
//     transition: @transition-basic;
//   }
//
//   &:hover {
//     &::before {
//       transform: translate(-50%, -50%) scale(1);
//       opacity: 1;
//     }
//   }
//   &:first-child {
//     margin-left: -7.1428%;
//   }
//   &.is-before-today {
//     color: @color-font-light;
//     pointer-events: none;
//   }
//   &.is-prev-month,
//   &.is-next-month {
//     color: @color-font-light;
//     font-weight: @font-weight-light;
//   }
//   &.is-today {
//     font-weight: @font-weight-medium;
//     color: @color-accent;
//   }
//
//   &.is-selected {
//     color: #ffffff;
//     font-weight: @font-weight-bold;
//     pointer-events: all;
//
//     &::before {
//       transform: translate(-50%, -50%) scale(1);
//       opacity: 1;
//       background-color: @color-accent;
//       border-color: @color-accent;
//     }
//     &:hover {
//       color: #000000;
//       font-weight: @font-weight-medium;
//       &::before {
//         background-color: transparent;
//         border: 1px dotted @color-accent;
//         opacity: 1;
//       }
//     }
//     &.is-prev-month,
//     &.is-next-month {
//       &::before {
//         opacity: 0.3;
//       }
//       &:hover::before {
//         opacity: 0.6;
//       }
//     }
//   }
// }
