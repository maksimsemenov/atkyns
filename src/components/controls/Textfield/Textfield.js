import React, { Component, PropTypes } from 'react'
import Autocomplete from 'react-autocomplete'
import Datefield from './Datefield'
import l from 'utils/local'
import './Textfield.less'


// Standart helper functions for autocomplete
export const renderMenu = (items, value, style) =>
  items ?
  <ul className='textfield-ac' children={items} style={style} /> :
  null

export const renderItem = (item, isHighlighted) =>
  <li
    key={item.value}
    className={`textfield-acItem${isHighlighted ? 'is-highlighted' : ''}`}
  >{item.name}</li>

export const getItemValue = item => item.name
export const shouldItemRender = (item, value) =>
  item.value.toLowerCase().search(value.toLowerCase()) !== -1

const defaultAC = {
  renderItem,
  renderMenu,
  getItemValue,
  shouldItemRender
}



/*
 * Textfield component itself
 */

const Textfield = ({
  value,
  transformIn = (value) => value,
  transformOut = (value) => value,
  wrapperStyle,
  appearance,
  autocomplete,
  onChange,
  onBlur = () => {},
  onFocus = () => {},
  error,
  note,
  ...rest
}) => {
  let ac = autocomplete ? {...defaultAC, ...autocomplete } : undefined
  let inputProps = {
    className: `textfield${error ? ' has-error' : ''}`,
    ...rest
  }
  return (
    <div className='textfield__control' style={wrapperStyle}>
      {ac ? (
        <Autocomplete
          value={transformIn(value)}
          onSelect={(value) => onChange(transformOut(value))}
          onChange={(event, value) => onChange(transformOut(value))}
          inputProps={inputProps}
          wrapperStyle={{ display: 'block' }}
          {...ac}
        />
        ) : (
        <input
          value={transformIn(value)}
          onChange={(e) => onChange(transformOut(e.target.value))}
          onBlur={(e) => onBlur(transformOut(e.target.value))}
          onFocus={(e) => onFocus(transformOut(e.target.value))}
          {...inputProps}
        />
      )}
      {error ? <div className='textfield__error'>{l(error)}</div> : null}
      {!error && note && <div className='textfield__note'>{l(note)}</div>}
    </div>
  )
}

Textfield.propTypes = {
  value: PropTypes.string,
  type: PropTypes.oneOf(['text', 'email', 'password', 'url', 'date', 'number']),
  wrapperStyle: PropTypes.object,
  autocomplete: PropTypes.shape({
    items: PropTypes.arrayOf(PropTypes.shape({
      value: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    })).isRequired,
    shouldItemRender: PropTypes.func
  }),
  transformIn: PropTypes.func,
  transformOut: PropTypes.func,
  onChange: PropTypes.func.isRequired,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,

  appearance: PropTypes.oneOf(['default', 'inGroup']),
}
Textfield.defaultProps = {
  type: 'text',
  onFocus: () => {},
  onBlur: () => {}
}

export default Textfield
