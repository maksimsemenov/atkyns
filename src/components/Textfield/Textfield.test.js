import React from 'react'
import ReactDOM from 'react-dom'
import Textfield from 'components/Textfield/Textfield'

describe('Textfield component', () => {
  it('renders without craches', () => {
    const div = document.createElement('div')
    const onChange = () => {}
    ReactDOM.render(<Textfield onChange={onChange} />, div)
  })
})
