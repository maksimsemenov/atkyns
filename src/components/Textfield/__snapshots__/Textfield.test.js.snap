exports[`Textfield component Snapshot test 1`] = `
<div
  className="textfield__control"
  style={undefined}>
  <input
    className="textfield"
    onChange={[Function onChange]}
    type="text"
    value={undefined} />
</div>
`;

exports[`Textfield component Snapshot test 2`] = `
<div
  className="textfield__control"
  style={undefined}>
  <input
    className="textfield"
    onChange={[Function onChange]}
    type="text"
    value={undefined} />
  <div
    className="ob-textfield-note">
    Note for the field
  </div>
</div>
`;

exports[`Textfield component Snapshot test 3`] = `
<div
  className="textfield__control"
  style={undefined}>
  <input
    className="textfield has-error"
    onChange={[Function onChange]}
    type="text"
    value={undefined} />
  <div
    className="ob-textfield-error">
    Error in field
  </div>
</div>
`;

exports[`Textfield component Snapshot test 4`] = `
<div
  className="textfield__control"
  style={undefined}>
  <div
    style={
      Object {
        "display": "inline-block"
      }
    }>
    <input
      aria-autocomplete="list"
      autoComplete="off"
      className="textfield"
      onBlur={[Function bound handleInputBlur]}
      onChange={[Function anonymous]}
      onClick={[Function bound handleInputClick]}
      onFocus={[Function bound handleInputFocus]}
      onKeyDown={[Function anonymous]}
      onKeyUp={[Function anonymous]}
      role="combobox"
      type="text"
      value="Test value" />
  </div>
</div>
`;

exports[`renderItem function Snapshot test 1`] = `
<li
  className="textfield-acItem">
  item2
</li>
`;

exports[`renderItem function Snapshot test 2`] = `
<li
  className="textfield-acItemis-highlighted">
  item2
</li>
`;

exports[`renderMenu function Snapshot test 1`] = `
<ul
  className="textfield-ac"
  style={undefined}>
  item1
  item2
</ul>
`;
